### ISSUE TITLE
Infinite compile time for `mixture-mpi.py` with `fusion` context  

### ISSUE THREAD

**Issue by MTCam** [illinois-ceesd/mirgecom#645]
The simple combustion example:  `mixture-mpi.py` fails to run in lazy mode with `fusion` array context.  Compile never completes, and eventually hangs my Mac laptop.

To reproduce, checkout `mirgecom@fusion_actx` and run:
> python -m mpi4py mixture-mpi.py --lazy

**Comment by matthiasdiener** [comment]
The issue seems to be that pymbolic's IdentityMapper has no caching. Working on a fix.

**Comment by matthiasdiener** [comment]
https://github.com/inducer/pymbolic/pull/92 fixes the long compile time but will need a whole bunch of changes in other packages

**Comment by inducer** [comment]
My understanding is that @kaushikcfd is pursuing something rather similar:

- https://github.com/inducer/pymbolic/pull/90
- https://github.com/inducer/loopy/pull/603

(Sorry!)

**Comment by kaushikcfd** [comment]
> but will need a whole bunch of changes in other packages

Yep, https://github.com/inducer/loopy/pull/603 contains fixes for that.
